src/lib/mina_lib/mina_lib.ml

```diff
@@ -401,6 +401,44 @@ let create_sync_status_observer ~logger ~demo_mode ~net
                     `Synced ) ) )
   in
   let observer = observe incremental_status in
+  (* monitor Mina status, issue a warning if offline for too long (unless we are a seed node) *)
+  ( if not is_seed then
+    let offline_timeout_min = 15.0 in
+    let offline_timeout_duration = Time.Span.of_min offline_timeout_min in
+    let offline_timeout = ref None in
+    let log_offline_warning _tm =
+      [%log warn] "Daemon has been continuously offline for %0.0f minutes"
+        offline_timeout_min
+    in
+    let start_offline_timeout () =
+      match !offline_timeout with
+      | Some _ ->
+          ()
+      | None ->
+          offline_timeout :=
+            Some
+              (Timeout.create () offline_timeout_duration
+                 ~f:log_offline_warning)
+    in
+    let stop_offline_timeout () =
+      match !offline_timeout with
+      | Some timeout ->
+          Timeout.cancel () timeout () ;
+          offline_timeout := None
```

Comment:
If the timeout had already triggered by this point, would be nice to give an info log saying that the node is now back online.